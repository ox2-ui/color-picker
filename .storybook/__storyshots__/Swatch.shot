exports[`coral`] = `
<div
  onClick={undefined}
  style={
    Object {
      "background": "#fff",
      "borderRadius": "1px",
      "boxShadow": "0 0 0 1px rgba(0,0,0,.1)",
      "cursor": "pointer",
      "display": "inline-block",
      "height": "20px",
      "padding": "5px",
      "width": "20px",
    }
  }>
  <div
    style={
      Object {
        "backgroundColor": "coral",
        "height": "100%",
        "width": "100%",
      }
    } />
</div>
`;

exports[`default`] = `
<div
  onClick={undefined}
  style={
    Object {
      "background": "#fff",
      "borderRadius": "1px",
      "boxShadow": "0 0 0 1px rgba(0,0,0,.1)",
      "cursor": "pointer",
      "display": "inline-block",
      "height": "20px",
      "padding": "5px",
      "width": "20px",
    }
  }>
  <div
    style={
      Object {
        "backgroundColor": undefined,
        "height": "100%",
        "width": "100%",
      }
    } />
</div>
`;

exports[`onClick`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "background": "#fff",
      "borderRadius": "1px",
      "boxShadow": "0 0 0 1px rgba(0,0,0,.1)",
      "cursor": "pointer",
      "display": "inline-block",
      "height": "20px",
      "padding": "5px",
      "width": "20px",
    }
  }>
  <div
    style={
      Object {
        "backgroundColor": "mediumvioletred",
        "height": "100%",
        "width": "100%",
      }
    } />
</div>
`;

exports[`size 50`] = `
<div
  onClick={undefined}
  style={
    Object {
      "background": "#fff",
      "borderRadius": "1px",
      "boxShadow": "0 0 0 1px rgba(0,0,0,.1)",
      "cursor": "pointer",
      "display": "inline-block",
      "height": "50px",
      "padding": "5px",
      "width": "50px",
    }
  }>
  <div
    style={
      Object {
        "backgroundColor": "coral",
        "height": "100%",
        "width": "100%",
      }
    } />
</div>
`;

exports[`yellowgreen`] = `
<div
  onClick={undefined}
  style={
    Object {
      "background": "#fff",
      "borderRadius": "1px",
      "boxShadow": "0 0 0 1px rgba(0,0,0,.1)",
      "cursor": "pointer",
      "display": "inline-block",
      "height": "20px",
      "padding": "5px",
      "width": "20px",
    }
  }>
  <div
    style={
      Object {
        "backgroundColor": "yellowgreen",
        "height": "100%",
        "width": "100%",
      }
    } />
</div>
`;
