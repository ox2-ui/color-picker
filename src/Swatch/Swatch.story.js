import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Swatch from './Swatch';

const handler = () => console.log('🍇'); // eslint-disable-line no-console

storiesOf('Swatch', module)
  .add('default', () => <Swatch />)
  .add('coral', () => <Swatch color="coral" />)
  .add('size 50', () => <Swatch color="coral" size={50} />)
  .add('onClick', () => (
    <Swatch color="mediumvioletred" onClick={handler} />
  ))
  .add('yellowgreen', () => <Swatch color="yellowgreen" />);
