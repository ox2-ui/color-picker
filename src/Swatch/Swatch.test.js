/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Swatch from './Swatch';

const handler = () => true;

it('renders correctly', () => {
  const wrapper = mount(
    <Swatch
      color="mediumvioletred"
      onClick={handler}
      size={50}
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
