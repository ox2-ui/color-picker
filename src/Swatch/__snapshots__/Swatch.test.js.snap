// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Swatch
  color="mediumvioletred"
  onClick={[Function]}
  size={50}
>
  <div
    onClick={[Function]}
    style={
      Object {
        "background": "#fff",
        "borderRadius": "1px",
        "boxShadow": "0 0 0 1px rgba(0,0,0,.1)",
        "cursor": "pointer",
        "display": "inline-block",
        "height": "50px",
        "padding": "5px",
        "width": "50px",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "mediumvioletred",
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </div>
</Swatch>
`;
